File: 백준/Silver/11399. ATM/ATM.py
import sys

input = sys.stdin.readline

N = int(input())
minutes = list(map(int, input().split()))

minutes.sort()
minutes_sum = []

M_sum = 0
for i in minutes:
    M_sum += i
    minutes_sum.append(M_sum)

print(sum(minutes_sum))

File: 백준/Silver/11399. ATM/ATM.js
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = require('fs').readFileSync(filePath).toString().trim().split('\n');

const N = Number(input.shift());
const P = input
  .shift()
  .split(' ')
  .map((v) => Number(v));

P.sort((a, b) => a - b);
P.unshift(0);
const dp = [...Array(N + 1).fill(0)];

for (let i = 1; i <= N; i++) {
  dp[i] = dp[i - 1] + P[i];
}
console.log(dp.reduce((acc, cur) => acc + cur, 0));
